kola/harness.go

```diff
@@ -112,11 +128,60 @@ func runTest(t Test, pltfrm string) (err error) {
 		fmt.Fprintf(os.Stderr, "%v instance up\n", pltfrm)
 	}
 
+	// drop kolet binary on machines
+	for _, m := range cluster.Machines() {
+		err = scpFile(m, "./kolet") //TODO pb: locate local binary path with `which` once kolet is in overlay
+		if err != nil {
+			return fmt.Errorf("dropping kolet binary: %v", err)
+		}
+
+		_, err := m.SSH("chmod +x ./kolet")
+		if err != nil {
+			return fmt.Errorf("chmod kolet binary: %v", err)
+		}
+	}
+
+	// close around t.Name
+	nativeRunner := func(funcName string, m platform.Machine) error {
+		// scp and execute kolet on remote machine
+		b, err := m.SSH(fmt.Sprintf("./kolet run %v %v", t.Name, funcName))
+		if err != nil {
+			return fmt.Errorf("%v: %s", err, b)
+		}
+		return nil
+	}
+
 	// run test
-	err = t.Run(cluster)
+	err = t.Run(cluster, nativeRunner)
 	return err
 }
 
+// scpFile copies file from src path to ~/ on machine
+func scpFile(m platform.Machine, src string) error {
+	in, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+	defer in.Close()
+
+	session, err := m.SSHSession()
+	if err != nil {
+		return fmt.Errorf("Error establishing ssh session: %v", err)
+	}
+	defer session.Close()
+
+	// machine reads file from stdin
+	session.Stdin = in
+
+	// cat file to fs
+	_, filename := filepath.Split(src)
+	_, err = session.CombinedOutput(fmt.Sprintf("cat > ./%s", filename))
```

Comment:
One trick I've used before is `"install -m 0755 /dev/stdin /dest/path"` and similar so you can set the mode at the same time.